### PR TITLE
Execute the oldest sync job first

### PR DESCRIPTION
**Pull request by wangch079**
This PR adds `sort` to `pending_jobs` of `SyncJobIndex` class, so that the oldest sync job is executed first.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/byoc.py [hunk 1]
```diff
@@ -710,7 +710,8 @@ async def pending_jobs(self, connector_ids):
                 ]
             }
         }
-        async for job in self.get_all_docs(query=query):
+        sort = [{"created_at": "asc"}]
```
Comment (timgrein) [inline connectors/byoc.py:713]: nit: "asc"/"desc" could be an enum, so we don't need to know the exact string value all the time, if we want to resue it

**Review by timgrein — APPROVED** [review]
One small nit, otherwise LGTM

File: connectors/byoc.py (hunk 1 above)
Comment (wangch079) [reply connectors/byoc.py]: Done: https://github.com/elastic/connectors-python/pull/793/commits/572276d742c9297d59f7e582d1805f66e618e8c9

**Review by timgrein — APPROVED** [review]
LGTM, thanks for incorporating the change

**Pull request merged**